File: couple-app-frontend/src/types/calendar.ts
export interface CalendarEvent {
  id: number;
  title: string;
  description?: string;
  event_date: string; // ISO date string
  event_type: 'anniversary' | 'milestone' | 'custom';
  couple_id: number;
  created_by: number;
  created_at: string;
  updated_at?: string;
}

export interface CalendarEventCreate {
  title: string;
  description?: string;
  event_date: string; // ISO date string
  event_type: 'anniversary' | 'milestone' | 'custom';
}

export interface CalendarEventUpdate {
  title?: string;
  description?: string;
  event_date?: string; // ISO date string
  event_type?: 'anniversary' | 'milestone' | 'custom';
}

export interface CoupleAnniversary {
  anniversary_date?: string; // ISO date string
}

File: couple-app/app/schemas/calendar.py
from pydantic import BaseModel, ConfigDict
from datetime import date, datetime
from typing import Optional


class CalendarEventBase(BaseModel):
    title: str
    description: Optional[str] = None
    event_date: date
    event_type: str  # 'anniversary', 'milestone', 'custom'


class CalendarEventCreate(CalendarEventBase):
    pass


class CalendarEventUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    event_date: Optional[date] = None
    event_type: Optional[str] = None


class CalendarEventResponse(CalendarEventBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    couple_id: int
    created_by: int
    created_at: datetime
    updated_at: Optional[datetime] = None


class CoupleAnniversaryUpdate(BaseModel):
    anniversary_date: Optional[date] = None
